src/Analyzers/CSharp/Tests/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementTests.cs

```diff
@@ -0,0 +1,814 @@
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestTrivia3()
+        {
+            var code =
+@"
+class C
+{
+    public C()
+    {
+        var v = true /*comment*/ [|?|]
+            0 :
+            1;
+    }
+}";
+
+            var fixedCode =
+@"
+class C
+{
+    public C()
+    {
+        var v = true /*comment*/
+            ? 0
+            : 1;
+    }
+}";
+
+            await new Verify.Test()
+            {
+                TestCode = code,
+                FixedCode = fixedCode,
+                Options = { { CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression, CodeStyleOptions2.FalseWithSuggestionEnforcement } }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestTrivia4()
+        {
+            var code =
+@"
+class C
+{
+    public C()
+    {
+        var v = true /*comment*/ [|?|] 
+            0 :
+            1;
+    }
+}";
+
+            var fixedCode =
+@"
+class C
+{
+    public C()
+    {
+        var v = true /*comment*/
+            ? 0
+            : 1;
+    }
+}";
+
+            await new Verify.Test()
+            {
+                TestCode = code,
+                FixedCode = fixedCode,
+                Options = { { CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression, CodeStyleOptions2.FalseWithSuggestionEnforcement } }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestTrivia5()
+        {
+            var code =
+@"
+class C
+{
+    public C()
+    {
+        var v = true /*comment1*/ [|?|] /*comment2*/
+            0 :
+            1;
+    }
+}";
+
+            var fixedCode =
+@"
+class C
+{
+    public C()
+    {
+        var v = true /*comment1*/ /*comment2*/
+            ? 0
+            : 1;
+    }
+}";
+
+            await new Verify.Test()
+            {
+                TestCode = code,
+                FixedCode = fixedCode,
+                Options = { { CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression, CodeStyleOptions2.FalseWithSuggestionEnforcement } }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestWithDiagnosticsElsewhere()
+        {
+            var code =
+@"
+class C
+{
+    public C(int{|CS1001:)|}
+    {
+        var v = true [|?|]
+            0 :
+            1;
+    }
+}";
+
+            var fixedCode =
+@"
+class C
+{
+    public C(int{|CS1001:)|}
+    {
+        var v = true
+            ? 0
+            : 1;
+    }
+}";
+
+            await new Verify.Test()
+            {
+                TestCode = code,
+                FixedCode = fixedCode,
+                Options = { { CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression, CodeStyleOptions2.FalseWithSuggestionEnforcement } }
+            }.RunAsync();
+        }
```

Comment:
Can we also add some FixAll in document/project/solution tests?

Reply:
there is a fix-all-in doc test.  i'm very 'meh' on the project/solution cases as it's basically just validating that this system can fix documents and merge the results :)